pkg/coscheduling/core/core.go

```diff
@@ -112,7 +119,46 @@ func (pgMgr *PodGroupManager) PreFilter(ctx context.Context, pod *corev1.Pod) er
 		return fmt.Errorf("cannot found engough pods, "+
 			"current pods number: %v, minMember of group: %v", len(pods), pg.Spec.MinMember)
 	}
-	return nil
+
+	// TODO(cwdsuzhou): This resource check can not absolutely to avoid unschedulable group.
+	// Only PreFilter resource constraints so even if a PodGroup passed here,
+	// it may not necessarily pass Filter due to other constraints such as affinity/taints.
+	nodes, err := pgMgr.snapshotSharedLister.NodeInfos().List()
+	if err != nil {
+		return err
+	}
+	if !pgMgr.enableResourcePreCheck {
+		return nil
+	}
+	if _, ok := pgMgr.permittedPG.Get(pgFullName); ok {
+		return nil
+	}
+	if pg.Spec.MinResources == nil {
+		matchCount := 0
+		for _, node := range nodes {
+			nodeCopy := node.Clone()
+			for matchCount < int(pg.Spec.MinMember) {
+				// we have to make sure node resources pre-filer called before this one.
+				errs := noderesources.Fits(pod, nodeCopy)
+				if len(errs) != 0 {
+					break
+				}
+				matchCount++
+				// Add pod to node to check if node can scheduler more than one pod
+				nodeCopy.AddPod(pod)
+			}
+			if matchCount >= int(pg.Spec.MinMember) {
+				pgMgr.permittedPG.Add(pgFullName, pgFullName, *pgMgr.scheduleTimeout)
+				return nil
+			}
+		}
+	} else {
+		clusterLeftResource := pgMgr.getLeftClusterResource()
+		if clusterResourceSatisfied(clusterLeftResource, framework.NewResource(*pg.Spec.MinResources)) {
```

Comment:
Two comments:

1. Combine the two methods, and return early when we walked to NodeX and have found that minResources are already satisfied.
1. Return a concrete error indicating which type of resource is not satisfied; instead of a binary true or false.

Reply:
BTW: I guess there is already some existing code snippet in k/k and hence can be reused here?

Reply:
> BTW: I guess there is already some existing code snippet in k/k and hence can be reused here?

I try to reuse the logic in K8s, but it seems hard to do this. Maybe we can expose some func in `Kubernetes` to do this. At this time, we can not re-use.